### ISSUE TITLE
Password Reset form doesn't work with UUID pks

### ISSUE THREAD

**Issue by yarbelk** [pennersr/django-allauth#994]
the block in ```accounts.utils.url_str_to_user_pk(s)``` is, as the comment states, Ugh.  with new UUID fields, it breaks on that test. to find its type.

NB - I think 1.8 moves to base 64 encoding

```
    def url_str_to_user_pk(s):
        User = get_user_model()
        # TODO: Ugh, isn't there a cleaner way to determine whether or not
        # the PK is a str-like field?
    try:
        User._meta.pk.to_python('a')
        pk = s
    except ValidationError:
        pk = base36_to_int(s)
    return pk
```

**Comment by pennersr** [comment]
Fixed by #995

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/994
